cmake/SanitizerOptions.cmake

```diff
@@ -0,0 +1,13 @@
+if(ENABLE_SANITIZER)
+  message(STATUS "Enabling address sanitizer (ASan).")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+endif()
```

Comment:
Per the [ASan](https://clang.llvm.org/docs/AddressSanitizer.html), to make the ASAN stacktraces easier to interpret, we should also add two additional flags: `-fno-omit-frame-pointer -fno-optimize-sibling-calls`.

Reply:
Done